cardano-crypto-class/src/Cardano/Crypto/DSIGN/EcdsaSecp256k1.hs

```diff
@@ -55,94 +80,194 @@ import Cardano.Crypto.DSIGN.Class (
+        psbUseAsSizedPtr psb $ \psp -> 
+          psbCreateSized $ \dstp ->
+            withForeignPtr secpCtxPtr $ \ctx -> 
+              void $ secpEcdsaSignatureSerializeCompact ctx dstp psp
+    {-# NOINLINE rawSerialiseVerKeyDSIGN #-}
+    rawSerialiseVerKeyDSIGN (VerKeyEcdsaSecp256k1 psb) = 
+      psbToByteString . unsafeDupablePerformIO . psbUseAsSizedPtr psb $ \psp ->
+        psbCreateLen @SECP256K1_ECDSA_PUBKEY_BYTES $ \ptr len -> do
+          let dstp = castPtr ptr
+          -- This is necessary because of how the C API handles checking writes:
+          -- maximum permissible length is given as a pointer, which is
+          -- overwritten to indicate the number of bytes we actually wrote; if
+          -- we get a mismatch, then the serialization failed. While an odd
+          -- choice, we have to go with it.
+          alloca $ \(lenPtr :: Ptr CSize) -> do
+            poke lenPtr len
+            withForeignPtr secpCtxPtr $ \ctx -> do
+              void $ secpEcPubkeySerialize ctx dstp lenPtr psp secpEcCompressed
+              writtenLen <- peek lenPtr
+              unless (writtenLen == len) 
+                     (error "rawSerializeVerKeyDSIGN: Did not write correct length for VerKeyDSIGN EcdsaSecp256k1DSIGN")
+    rawSerialiseSignKeyDSIGN (SignKeyEcdsaSecp256k1 psb) = psbToByteString psb
+    {-# NOINLINE rawDeserialiseSigDSIGN #-}
+    rawDeserialiseSigDSIGN bs = 
+      SigEcdsaSecp256k1 <$> (psbFromByteStringCheck bs >>= go)
+      where
+        go :: 
+          PinnedSizedBytes SECP256K1_ECDSA_SIGNATURE_BYTES -> 
+          Maybe (PinnedSizedBytes SECP256K1_ECDSA_SIGNATURE_BYTES_INTERNAL)
+        go psb = unsafeDupablePerformIO . psbUseAsSizedPtr psb $ \psp -> do
+          (sigPsb, res) <- psbCreateSizedResult $ \sigp ->
+            withForeignPtr secpCtxPtr $ \ctx -> 
+              secpEcdsaSignatureParseCompact ctx sigp psp
+          pure $ case res of 
+            1 -> pure sigPsb
+            _ -> Nothing
+    {-# NOINLINE rawDeserialiseVerKeyDSIGN #-}
+    rawDeserialiseVerKeyDSIGN bs = 
+      VerKeyEcdsaSecp256k1 <$> (psbFromByteStringCheck bs >>= go)
+      where
+        go :: 
+          PinnedSizedBytes SECP256K1_ECDSA_PUBKEY_BYTES -> 
+          Maybe (PinnedSizedBytes SECP256K1_ECDSA_PUBKEY_BYTES_INTERNAL)
+        go psb = unsafeDupablePerformIO . psbUseAsCPtrLen psb $ \p srcLen -> do
+          let srcp = castPtr p
+          (vkPsb, res) <- psbCreateSizedResult $ \vkp ->
+            withForeignPtr secpCtxPtr $ \ctx -> 
+              secpEcPubkeyParse ctx vkp srcp srcLen
+          pure $ case res of 
+            1 -> pure vkPsb
+            _ -> Nothing
+    rawDeserialiseSignKeyDSIGN bs = 
+      SignKeyEcdsaSecp256k1 <$> psbFromByteStringCheck bs
 
 instance ToCBOR (VerKeyDSIGN EcdsaSecp256k1DSIGN) where
-  toCBOR = encodeVerKeyDSIGN
-  encodedSizeExpr _ = encodedVerKeyDSIGNSizeExpr
+    toCBOR = encodeVerKeyDSIGN
+    encodedSizeExpr _ = encodedVerKeyDSIGNSizeExpr
```

Comment:
Almost forgot, what happened to this indentation and instances below as well, why did it become 4 spaces?

Reply:
That's a holdover from something I later removed. Will fix.